mitodl/mitxpro
# #324 Home Page: Base

#### Pre-Flight checklist


#### What are the relevant tickets?
#324 
#335 

#### What's this PR do?
Adds a base `HomePage` model so that a custom home page may be added to Wagtail. Also makes some minor changes to existing partials that make `Courseware Carousel`, `Text-Image Section`, `Icon Grid` and `Testimonial Section` compatible with the home page.

#### How should this be manually tested?
Create a new page of the type `HomePage`. Go into the settings of `Wagtail CMS admin > Settings > Sites > (Default Site)` and set this new page as the `Root page`. Also, make sure that the `hostname` and `port` properties are setup correctly. Next, go to the root of your deployment and you should see the blank home page with the `top-app-header` and `footer` at the bottom. If you add any subpages to this home page, these sections should start appearing on the page.

**Note:** To have correct "live page url" working for all course/program pages, move them under your new home page. Ideally you should have all your existing pages under the new home page and delete the old, default home page entry.

~~**Note:** This PR adds a migration to automatically move existing pages to our custom HomePage instance.~~

~~**N.B.:** After some discussion with @rhysyngsun and searching the Wagtail knowledgebase/repository it is evident that a migration is not the place to do page tree modifications. This PR includes a management command `configure_home_page` that can also be reverted with the optional `--uninstall` parameter. This command will move all existing pages under the new `HomePage` instance and also make this instance the root of the default `Site`.~~

#### Screenshots (if appropriate)
![Screenshot from 2019-05-21 15-35-26](https://user-images.githubusercontent.com/45350418/58089418-1e21d280-7bde-11e9-92b2-4ad40dd89f81.png)
![Screenshot from 2019-05-21 15-36-43](https://user-images.githubusercontent.com/45350418/58089484-51646180-7bde-11e9-81fa-4b8649a12ac5.png)
